src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,65 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { ChangeEvent } from 'react';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const possibleCenturies = ['16', '17', '18', '19', '20'];
+  const centuriesParams = searchParams.getAll('centuries') || [];
+
+  const onSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const query = { query: e.target.value || null };
+
+    setSearchParams(getSearchWith(searchParams, query));
+  };
+
+  const setCenturies = (century: string) => {
+    if (centuriesParams.includes(century)) {
+      const updatedCentury = centuriesParams.filter(c => c !== century);
+
+      return { centuries: updatedCentury };
+    }
+
+    return { centuries: [...centuriesParams, century] };
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
-      <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+      <p
+        className="panel-tabs"
+        data-cy="SexFilter"
+      >
+        <Link
+          className={cn({
+            'is-active': searchParams.get('sex') === null,
+          })}
+          to="/people"
+        >
+          All
+        </Link>
+        <Link
+          className={cn({
+            'is-active': searchParams.get('sex') === 'm',
+          })}
+          to={{
+            search: getSearchWith(searchParams, { sex: 'm' }),
+          }}
+        >
```

Comment:
You repeated this logic several times.
Haven't you thought about creating a custom component 
`SearchLink`?
```jsx
<SearchLink
  params={{ sex: 'm' }}
  className={cn({ 'is-active': sex === 'm' })}
>
  All
</SearchLink>